### PR TITLE
Create mapping from NotificationOption to EditorConfig severity string.

### PR DESCRIPTION
**Pull request by JoeRobich**
This PR is being created to simplify #30760

`notificationOption.ToString().ToLowerInvariant();` would return a localized string instead of the correct severity string.

### CODE REVIEW

**Comment by JoeRobich** [comment]
@sharwell Please take a look

**Comment by JoeRobich** [comment]
@vatsalyaagrawal for approval

File: src/Workspaces/Core/Portable/Extensions/NotificationOptionExtensions.cs [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
+            return notificationOption.Severity switch
             {
-                return notificationOption.ToString().ToLowerInvariant();
-            }
+                ReportDiagnostic.Suppress => EditorConfigSeverityStrings.None,
+                ReportDiagnostic.Hidden => EditorConfigSeverityStrings.Silent,
+                ReportDiagnostic.Info => EditorConfigSeverityStrings.Suggestion,
+                ReportDiagnostic.Warn => EditorConfigSeverityStrings.Warning,
+                ReportDiagnostic.Error => EditorConfigSeverityStrings.Error,
+                _ => throw ExceptionUtilities.Unreachable
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Extensions/NotificationOptionExtensions.cs]: There's also ExceptionUtilities.UnexpectedValue() for things like enum switches.

**Comment by JoeRobich** [comment]
Rebased on dev16.1-preview2

**Comment by JoeRobich** [comment]
@vatsalyaagrawal for approval

**Pull request merged**